radio/src/storage/yaml/yaml_datastructs_funcs.cpp

```diff
@@ -94,15 +87,21 @@ static uint32_t in_read_weight(const YamlNode* node, const char* val, uint8_t va
 bool in_write_weight(const YamlNode* node, uint32_t val, yaml_writer_func wf,
                      void* opaque)
 {
-  int32_t sval = yaml_to_signed(val, node->size <= 11 ? node->size : 11);
-  int32_t gvar = (node->size > 8 ? GV1_LARGE : GV1_SMALL);
+  int32_t sval = yaml_to_signed(val, node->size);
 
-  if (sval >= gvar - 10 && sval <= gvar) {
-    char n = gvar - sval + '0';
-    return wf(opaque, "-GV", 3) && wf(opaque, &n, 1);
-  } else if (sval <= -gvar + 10 && sval >= -gvar) {
-    char n = val - gvar + '1';
-    return wf(opaque, "GV", 2) && wf(opaque, &n, 1);
+  if (GV_IS_GV_VALUE(sval)) {
+    char s[8] = "";
+    int ofst = 0;
+    char idx = GV_INDEX_FROM_VALUE(sval);
+    if (idx < 0) {
+      s[0] = '-';
+      ofst = 1;
+      idx = -idx;
+    } else {
+      idx = idx + 1;
+    }
+    strAppendStringWithIndex(s + ofst, "GV", idx);
```

Comment:
Whereby very unlikely, as this function doesn't check the buffer length, this introduces a potential overflow.

Reply:
As this is writing data from the radio (not reading from a file), and there is no possibility of ever having 10,000 global variables, I think it is pretty safe.